src/PageLayout/Notification/NotificationDetailsController.js

```diff
@@ -59,137 +61,104 @@ class NotificationDetailsController extends Component {
-                notes: 'Enter through the back door.',
-                umbrella: 'RheaQY1WxJT03sTPQICFZ4STpfm1',  // uid-key of a umbrella
-                donatingAgency: '-K9HdKlCLjjk_ka82K0s',  // autogen-key of a donating-agency
-                requester: 'Andrea Benson',  // name of a donating-agency-member
-                receivingAgency: {
-                    // Depending on the status of the request, this
-                    // map will have 1 out of the 3 possible fields:
-                    // "requested", "pending" (list), "claimed"
-                    requested: 'uGOFJ8NqHjbZhKAYzSZFRs1dSKD3',  // uid-key of a RA
-                    // OR
-                    pending: [  
-                        'uGOFJ8NqHjbZhKAYzSZFRs1dSKD3',  // uid-key of all RAs
-                    ],
-                    // OR
-                    claimed: 'uGOFJ8NqHjbZhKAYzSZFRs1dSKD3',  // uid-key of a RA (once a RA claims)
-                },
-                // ADDED FOR DUMMY DATA
-                daInfo: {
-                    name: 'Local Point',
-                    primaryContact: { // primaryContact of the request not the DA
-                        name: 'Andrea Benson',
-                        email: 'AndreaIsCool@uw.edu',
-                        phone: '206-586-9876',
-                        position: 'Manager'
-                    },
-                    address: {
-                        street1: '1201 NE Campus Pkwy',
-                        street2: '',
-                        city: 'Seattle',
-                        state: 'WA',
-                        zipcode: 98105,
-                        officeNo: '220'
+            let genSupplementaryPromise = (pathPrefix, accountId) => (new Promise(async (resolve, reject) =>
+                resolve((await db.ref(`${pathPrefix}/${accountId}`).once('value')).val())));
+
+            db.ref(`delivery_requests/${this.props.account.umbrella}/${this.state.notification.content}`)
+                .once('value', async (snap) => {
+                    let deliveryRequest = snap.val();
+                    // kick all promises off
+                    let donatingAgencyPromise = genSupplementaryPromise('donating_agencies', deliveryRequest.donatingAgency);
+                    let donatingAgencyContactPromise = genSupplementaryPromise('accounts', deliveryRequest.primaryContact);
+                    let delivererGroupPromise = (deliveryRequest.delivererGroup && deliveryRequest.delivererGroup.claimed) ?
+                        genSupplementaryPromise('accounts', deliveryRequest.delivererGroup.claimed) : null; 
+                    let receivingAgencyPromise = (deliveryRequest.receivingAgency && deliveryRequest.receivingAgency.claimed) ? 
+                        genSupplementaryPromise('accounts', deliveryRequest.receivingAgency.claimed) : null;
+
+                    // collect them
+                    let donatingAgency = await donatingAgencyPromise;
```

Comment:
The promises are already running, so the await keyword just rejoins the processes. If we're waiting on the `donatingAgency`, the asynchronous processes getting `donatingAgencyContact` and `delivererGroup` are still executing. Since the cost of actually returning the result of the resolved promise is negligible, the total runtime then becomes `max(cost_of_each_async_process) + sum(cost_of_joining_each_process)`, which is the same as doing `Promise.all`.

I think the benefit of doing it this way is we can load each result into its own variable, instead of returning a list from `Promise.all` and then having to explode (I guess we could do like `let [donatingAgency, donatingAgencyContact...] = Promise.all([promise1, promise2...])` but I think each on its own line is cleaner, especially since some of the promises may be undefined.